### ISSUE TITLE
Installation fails due to rate limits

### ISSUE THREAD

**Issue by ctron** [argoproj/argo-cd#4917]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

The installation fails due to rate limits on Docker Hub:

~~~
Failed to pull image "argoproj/argocd@sha256:b835999eb5cf75d01a2678cd971095926d9c2566c9ffe746d04b83a6a0a2849f": rpc error: code = Unknown desc = Error reading manifest sha256:b835999eb5cf75d01a2678cd971095926d9c2566c9ffe746d04b83a6a0a2849f in docker.io/argoproj/argocd: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
~~~

**To Reproduce**

Pull images.

**Expected behavior**

It would be great to provide an alternate location of the container images, which doesn't have such tight rate limits.

**Screenshots**

See above

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```Failed to pull image "argoproj/argocd@sha256:b835999eb5cf75d01a2678cd971095926d9c2566c9ffe746d04b83a6a0a2849f": rpc error: code = Unknown desc = Error reading manifest sha256:b835999eb5cf75d01a2678cd971095926d9c2566c9ffe746d04b83a6a0a2849f in docker.io/argoproj/argocd: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

**Comment by jannfis** [comment]
Agree, this is bad. We are currently evaluating a target registry for migration of the images.

/cc @jessesuen @alexmt @edlee2121

**Comment by edlee2121** [comment]
FYI, I just submitted an application to have argoproj exempted from the limit.

![image](https://user-images.githubusercontent.com/13398326/100400665-2c8b4e80-300c-11eb-983a-d1ad2f15f9f0.png)

**Comment by jessesuen** [comment]
@edlee2121 shall we start moving all our images to google container registry and changing default instal manifests to use it?

Another option is github container registry, although it's a bit unproven.

**Comment by edlee2121** [comment]
@jessesuen That's not a bad option. Your call.

**Comment by ctron** [comment]
GitHub container repository (https://ghcr.io) works quite well, as does https://quay.io

**Comment by grtrout** [comment]
Have any alternate registries been identified and/or used? Like many others, I am part of a large organization and, as a result, the pull rate limit seems to _always_ be a problem. I decided to do a full cluster reinstall an hour or so ago and am dead in water at this step, waiting for some indeterminate amount of time. I'm not sure if upgrading to Docker Pro/Team is a feasible option for me. In the meantime, I will self-host the latest stable images as soon as I can pull them...

**Comment by alexmt** [comment]
hello @grtrout , 

Argo CD moved to quay.io: `quay.io/argoproj/argocd`

**Comment by grtrout** [comment]
> Argo CD moved to quay.io: `quay.io/argoproj/argocd`

I have not used quay before. Is there a documented way to use that repo instead of Docker Hub? I am installing based on the resource definition in the quick start guide. I.e.: 
https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml

**Comment by sbose78** [comment]
@grtrout The user experience should be unchanged.

**Comment by gasgithub** [comment]
If you have existing installation managed by operator, which fails to run due to rate limits you will have to go to your ArgoCD operator, edit Argo CD resource and add the following in spec:
```
spec:
  image: quay.io/argoproj/argocd
```
Otherwise it constantly tries to reach Dockerhub.

**Comment by saggar** [comment]
Hi. We're facing Docker rate limiting with our KubeFlow pipelines on AWS EKS. Is there a way for users to specify the ```wait``` container image (here ```argoproj/argoexec:v2.3.0```) when submitting a pipeline step? This is something a user can do for the ```main``` container. However, couldn't find an option for the ```wait``` container.

```
    involvedObject:
      kind: Pod
      namespace: kubeflow-admin
      name: multi-z5568-154393337
      uid: 6cf5c442-72ae-4fe2-b891-28a37ad70f65
      apiVersion: v1
      resourceVersion: '58717425'
      fieldPath: 'spec.containers{wait}'
    reason: Failed
    message: >-
      Failed to pull image "argoproj/argoexec:v2.3.0": rpc error: code = Unknown
      desc = Error response from daemon: toomanyrequests: You have reached your
      pull rate limit. You may increase the limit by authenticating and
      upgrading: https://www.docker.com/increase-rate-limit
```

**Comment by wtam2018** [comment]
@saggar If you pull the image from quay.io/argoproj/argoexec repository, you won't have rate limit issue.
